src/main/java/com/direwolf20/buildinggadgets/common/concurrent/TransactionPoolExecutor.java

```diff
@@ -0,0 +1,94 @@
+package com.direwolf20.buildinggadgets.common.concurrent;
+
+import com.direwolf20.buildinggadgets.api.building.view.IBuildContext;
+import com.direwolf20.buildinggadgets.api.exceptions.TransactionExecutionException;
+import com.direwolf20.buildinggadgets.api.template.ITemplate;
+import com.direwolf20.buildinggadgets.api.template.transaction.ITemplateTransaction;
+import com.direwolf20.buildinggadgets.common.BuildingGadgets;
+
+import javax.annotation.Nullable;
+import java.util.concurrent.RejectedExecutionException;
+import java.util.concurrent.SynchronousQueue;
+import java.util.concurrent.ThreadPoolExecutor;
+import java.util.concurrent.TimeUnit;
+import java.util.function.BiConsumer;
+import java.util.function.Consumer;
+
+public enum TransactionPoolExecutor {
```

Comment:
Just a question, I might be wrong. Isn't there already a PoolExecutor in java?

Reply:
I don't fully understand what you are trying to say with that.

Yes there are ThreadPoolExecutors... That's why I instantiate one...?

Reply:
Yup I'm a fucking idiot. Ignore me :D